pkgs/by-name/do/dorion/package.nix

```
@@ -1,65 +1,176 @@
 { lib
 , stdenv
+, fetchFromGitHub
+, fetchYarnDeps
 , fetchurl
+, rustPlatform
 , autoPatchelfHook
-, dpkg
+, wrapGAppsHook
+, esbuild
 , glib-networking
 , gst_all_1
 , libappindicator
 , libayatana-appindicator
+, nodejs
+, openssl
+, pkg-config
+, pnpm-lock-export
+, prefetch-yarn-deps
 , webkitgtk
-, wrapGAppsHook
+, yarn
 }:
 
-stdenv.mkDerivation (finalAttrs: {
-  name = "dorion";
-  version = "1.2.1";
+let
+
+  pname = "dorion";
+  version = "2.2.0";
 
-  src = fetchurl {
-    url = "https://github.com/SpikeHD/Dorion/releases/download/v${finalAttrs.version }/Dorion_${finalAttrs.version}_amd64.deb";
-    hash = "sha256-FghJM34GMt8+4b6jsQQSsfmHIyua/pjRHKNErGyK/kw=";
+  src = fetchFromGitHub {
+    owner = "SpikeHD";
+    repo = pname;
+    rev = "08f784baa85bbdc2a0125a6133d85827a8d891dc";
+    sha256 = "sha256-6lYnRW1vHmwuNEr2NNgyPtq/m4dQNmBVBquje528IGI=";
+
+    nativeBuildInputs = [ pnpm-lock-export ];
+    postFetch = ''
+      (cd $out && pnpm-lock-export --schema yarn.lock@v1)
+    '';
   };
 
-  unpackCmd = ''
-    dpkg -X $curSrc .
-  '';
+  shelter = fetchurl {
+    url = "https://raw.githubusercontent.com/uwu/shelter-builds/ea1db7b8780b56b9dabf65a0ea7bd87a0265907e/shelter.js";
+    hash = "sha256-pH+zc7ck6rP3F3UQZHTtORYYvEH6HAbI2bJYcKaCFUU=";
+  };
```

This fetch struck me as a backdoor.
```suggestion
    meta.homepage = "https://github.com/uwu/shelter";
    meta.sourceProvenance = with sourceTypes; [ binaryNativeCode ];
  };
```

---

if possible, could it be built from source? not a blocker

---

Set `meta.license` as well

---

This is not `binaryNativeCode`, though I suppose we don't have a good source type right now to describe prebuilt/minified typescript

If we need to put something here, I suppose `binaryBytecode` would be closest?

Looking into it, a source build could be done in a separate derivation, but would be prohibitively difficult until we have pnpm-native tooling given its use of pnpm workspaces

---

To be clear, going by the comment below, the "built from source" is not a blocker, but i consider the inclusion of a `meta` section to be one. This is to my understanding a package, which by default is unfree unless permission/a license is explicitly given.

---

Added the license